approx_dates/models.py

```diff
@@ -26,8 +26,34 @@ def __init__(self, earliest_date, latest_date, source_string=None):
         self.latest_date = latest_date
         self.source_string = source_string
 
+    def to_iso8601(self):
+        if self.earliest_date == self.latest_date:
+            return self.earliest_date.isoformat()
+        else:
+        #is it a vague month?
```

Comment:
For this part of the method, I'd be strongly inclined to factor out two new methods:

`def is_partial_just_year(self)`

and

`def is_partial_just_year_and_month(self)`

I suspect these are likely to be more generally useful, and it'll make this method a bit clearer.